packages/cli-lib/lib/config.js

```diff
@@ -420,20 +420,24 @@ const getAccountId = nameOrId => {
   let accountId;
   let account;
 
+  const setNameOrAccountFromSuppliedValue = suppliedValue => {
+    if (typeof suppliedValue === 'number') {
+      accountId = suppliedValue;
+    } else if (/^\d+$/.test(suppliedValue)) {
+      accountId = parseInt(suppliedValue, 10);
+    } else {
+      name = suppliedValue;
+    }
+  };
+
   if (!nameOrId) {
     const defaultAccount = getConfigDefaultAccount(config);
 
     if (defaultAccount) {
-      name = defaultAccount;
+      setNameOrAccountFromSuppliedValue(defaultAccount);
```

Comment:
I think it is worth adding test coverage for this change in behavior.

Reply:
Good call. Added a test for this case.